Mono.Debugging/Mono.Debugging.Client/DebuggerSession.cs

```diff
@@ -297,7 +297,12 @@ void Dispatch (Action action)
 									}
 								}
 								lock (slock) {
-									actionToExecute ();
+									try {
+										actionToExecute ();
+									} catch (Exception ex) {
+										if (!HandleException (ex))
+											throw;
```

Comment:
You can't throw!! this will terminate the IDE.

Reply:
oops